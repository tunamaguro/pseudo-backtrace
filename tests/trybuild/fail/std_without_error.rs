use pseudo_backtrace::StackError;

#[derive(Debug)]
struct NotError;
impl core::fmt::Display for NotError {
    fn fmt(&self, f: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
        "not-error".fmt(f)
    }
}
// no core::error::Error impl

#[derive(Debug, StackError)]
struct BadOuter {
    #[stack_error(std)]
    source: NotError,
    #[location]
    location: &'static core::panic::Location<'static>,
}

impl core::fmt::Display for BadOuter {
    fn fmt(&self, f: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
        "bad".fmt(f)
    }
}
impl core::error::Error for BadOuter {}

fn main() {}
