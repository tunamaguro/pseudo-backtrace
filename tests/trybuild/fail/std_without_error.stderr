error[E0599]: the method `as_dyn_std_error` exists for struct `NotError`, but its trait bounds were not satisfied
  --> tests/trybuild/fail/std_without_error.rs:12:17
   |
 4 | struct NotError;
   | --------------- method `as_dyn_std_error` not found for this struct because it doesn't satisfy `NotError: AsDynStdError<'_>` or `NotError: std::error::Error`
...
12 | #[derive(Debug, StackError)]
   |                 ^^^^^^^^^^ method cannot be called on `NotError` due to unsatisfied trait bounds
   |
   = note: the following trait bounds were not satisfied:
           `NotError: std::error::Error`
           which is required by `NotError: AsDynStdError<'_>`
note: the trait `std::error::Error` must be implemented
  --> $RUST/core/src/error.rs
   |
   | pub trait Error: Debug + Display {
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   = help: items from traits can only be used if the trait is implemented and in scope
   = note: the following trait defines an item `as_dyn_std_error`, perhaps you need to implement it:
           candidate #1: `AsDynStdError`
   = note: this error originates in the derive macro `StackError` (in Nightly builds, run with -Z macro-backtrace for more info)
